src/components/ErrorBoundary.tsx

```diff
@@ -1,30 +1,48 @@
 import * as Sentry from '@sentry/react';
 import React from 'react';
-import { Error } from './Error';
+import { Error as ErrorComponent, ErrorPropTypes } from './Error';
 import type { ErrorBoundaryProps } from '@sentry/react/types/errorboundary';
 import { ErrorContext } from '../contexts';
 import { SentryError } from '../types';
+import { SessionExpiredError } from '@openstax/ts-utils/errors';
+
+const Error = ({ children, ...props }: React.PropsWithChildren<ErrorPropTypes>) =>
+  <ErrorComponent data-testid='error-fallback' {...props}>{children}</ErrorComponent>;
+
+export const defaultErrorFallbacks = {
+  'generic': <Error data-testid='error-fallback' />,
+  [SessionExpiredError.TYPE]: <Error heading='Your session has expired'>
+    Please refresh your browser and try again.
+  </Error>,
+};
+
+const getTypeFromError = (error: Error | PromiseRejectionEvent['reason']) =>
+  'TYPE' in error.constructor && typeof error.constructor.TYPE === 'string' ?
+  error.constructor.TYPE : undefined;
```

Comment:
i think in this library we can avoid the dependency on `ts-utils` by using `error.constructor.name` for subtypes of error. that would allow someone not using ts-utils to still benefit from using this error boundary

Reply:
That sounds good. I wasn't sure if we wanted a hard dependency on it just to we can do `SessionExpiredError.TYPE` for keys. Probably not worrying about for our use case, I'll update it to a plain string. I think I'll need to check `error.constructor.TYPE` though? When I tested the [deployed code](https://openstax.github.io/ui-components/refs/heads/special-error-components/7d978feb446b201eb87d3fe94257d7fb87c6aaaa/?story=error-boundary--fallback--special-error) `name` gets minified into two letters. Maybe we should change the extended classes to override constructor and set `name`?

Reply:
Maybe this works to fallback on the name:
```typescript
  const { TYPE, name } = error.constructor;
  return TYPE && typeof TYPE === 'string' ? TYPE : name;
```